### ISSUE TITLE
Tracking issues for blog posts/articles to accompany the MirageOS 4 release

### ISSUE THREAD

**Issue by samoht** [mirage/mirage-www#704]
Similar to to #513 but for MirageOS 4. I am opening this issue to organize our communication around the upcoming release.

Here is the current plan (days are tentative):
- Day 1: 
  - Linux Foundation press release 
  - Mirage 4.0 release announcement (we want to dedicate the release to [Lars](https://xenproject.org/2020/01/31/saying-goodbye-to-lars-kurth-open-source-advocate-and-friend/))
- Day 3
  - ["Email in OCaml" (Draft)](https://hackmd.io/tBGmh_ogRkWsXAkfYyS0Ug) @dinosaure 
- Day5
  - ["Introduction to Build Concepts in MirageOS 4.0" (Draft)](https://hackmd.io/-EQhtniuTzeN_sghMbVBSA) @TheLortex 
- Day 7
  -  ["OS as a Development Journey" (Draft)](https://hackmd.io/7IH5SE6xRXaHqBLkuvuQKw) (Pierre)
- Day 9
  - post on VPNKit (pointing to the [the Docker blog post](https://www.docker.com/blog/how-docker-desktop-networking-works-under-the-hood/) (@djs55)
-  Day 11
    - summary post on Robur's unikernels (just a recap of the unikernel section in [an old post](https://hannes.nqsb.io/Posts/NGI) + the new DAPSI unikernels? (@hannesm + @dinosaure )

**Comment by hannesm** [comment]
The issue with "something about reproducible builds" is that for MirageOS 3.x we have everything in place (https://hannes.robur.coop/Posts/Deploy https://builds.robur.coop) -- while the story for MirageOS 4 is not yet clear (I discussed with @dinosaure to go more in depth in Mirleft in November with @TheLortex and @dinosaure).

**Comment by samoht** [comment]
Ok no pb! Anything else you would be interested to write for the release? Maybe a summary of the nice unikernel apps that you've been developing at Robur (like OpenVPN?)? (or something about DNS which might be timely given the recent Slack and FB outages? :p)

**Comment by dinosaure** [comment]
I think it should be a good place to talk about `mirage-crypto` too:
- the incredible work from @hannesm about its integration of [`fiat`][] into `mirage-crypto`
- our work on `eqaf`
- and the story behind `mirage-crypto` generally which is a big package in our ecosystem

Then, I agree with @samoht to talk about `ocaml-dns` since we started to deeply use it (if only for our domain name).

Finally, it can be an opportunity to talk more precisely about `albatross` & `ocurrent` indeed (and explain that the story to deploy a MirageOS is more easier than before).

[`fiat`]: https://github.com/mit-plv/fiat-crypto

**Comment by dinosaure** [comment]
Ah and may be we missed it but I think it improved a lot our ecosystem but we should talk about `ocaml-ci` (@avsm, @talex5) as our default CI for any of our MirageOS projects. It solves many issues and specially big-endian support and 32-bits architecture support which, IMHO, is in the spirit of MirageOS projects where our software should work everywhere.

**Comment by dinosaure** [comment]
And (again), we should talk a bit about the IPv6 support too I think (@MagnusS & @hannesm?).

**Comment by samoht** [comment]
We should aim to at most 4 or 5 blog posts - it's fine for these to be quite short, and point to existing materials. The goals of these posts are to showcase existing work and give their context. So we can probably merge a few things together if that's easier.

For design overviews/architecture documents, it's better to create a separate entry in the wiki/tutorial guides (that we can keep up-to-date in the future) and refer to them in the overview blog posts.

**Comment by hannesm** [comment]
> talk a bit about the IPv6 support

This is already announced in https://mirage.io/blog/announcing-mirage-310-release

**Comment by hannesm** [comment]
> OpenVPN

sure, although there's some gap of it still not released to opam

> DNS

there are some points to DNS (none of which are really related to MirageOS 4) that are still under development:
- DNS client using multiple nameservers (just merged that PR) via happy-eyeballs
- DNS-over-TLS (WIP https://github.com/mirage/ocaml-dns/pull/267)
- DNSSec (not done yet, see https://github.com/mirage/ocaml-dns/issues/236 https://github.com/mirage/ocaml-dns/pull/251)
- IPv6 support in the resolver & server (not quite done, see https://github.com/mirage/ocaml-dns/issues/262 https://github.com/mirage/ocaml-dns/issues/253)

TL;DR: I'd wait until we've a DNS release and then announce the features implemented. Indeed, a DNS resolver that validates DNSSec is still high on my priority list.

**Comment by hannesm** [comment]
> mirage-crypto (and EC)

See https://hannes.robur.coop/Posts/EC for my writeup in that respect. Feel free to link to that blog post as well ;)

Note: I'm happy to write a blog post, but atm I'm not entirely sure what about. Eventually the builds.robur and albatross integration -- but then this will have to wait until builds.robur uses MirageOS4. Eventually deployment with albatross + DHCP + binary builds, but this needs to wait until that prototype is done.

**Comment by hannesm** [comment]
Eventually the CA https://github.com/paurkedal/ephca is also a nice use case, and the sshfs https://github.com/palainp/mirage-sshfs (I have tested neither of them and am not sure how early-development they are).

**Comment by dinosaure** [comment]
> what is the state of a mail server

Currently, we are able to send an email with DKIM/SPF mechanism and a service such as Google can receive emails and does not throw them as a spam. I can share in private some outcomes about the current status of such stack if you want.

**Comment by dinosaure** [comment]
@TheLortex and me probably want to write down something about `gilbraltar` when we discover many interesting things (the complexity of a bare-metal Operating System according our cross-compilation workflow, the signal system and so on). WDYT?

**Comment by hannesm** [comment]
Thanks for the organisation and the release to everyone, especially @samoht @dinosaure @christinerose @TheLortex. I suspect this issue is done and will close it. If there's something left, please reopen. :)

**Comment by christinerose** [comment]
You're welcome, @hannesm! It was my pleasure to work with this team. 😀

**Comment by samoht** [comment]
Many thanks to everyone involved, we managed to publish great content for 2 weeks without missing any slot! 🎉 And many many thanks to @christinerose to coordinate all of this :-)
